core/namespace/service.go

```diff
@@ -0,0 +1,68 @@
+package namespace
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"github.com/odpf/salt/log"
+)
+
+type Service struct {
+	storageRepo   StorageRepository
+	discoveryRepo DiscoveryRepository
+	logger        log.Logger
+}
+
+func NewService(logger log.Logger, storageRepo StorageRepository, discoveryRepo DiscoveryRepository) *Service {
+	return &Service{
+		logger:        logger,
+		storageRepo:   storageRepo,
+		discoveryRepo: discoveryRepo,
+	}
+}
+
+func (s Service) MigrateDefault(ctx context.Context) (string, error) {
+	return s.Create(ctx, DefaultNamespace)
+}
+
+func (s Service) Create(ctx context.Context, namespace *Namespace) (string, error) {
+	id, err := s.storageRepo.Create(ctx, namespace)
+	if err != nil {
+		return "", err
+	}
+	if err := s.discoveryRepo.CreateNamespace(ctx, namespace); err != nil {
+		return "", err
+	}
+	return id, nil
+}
+
+// Update can't modify a namespace ID and name
+func (s Service) Update(ctx context.Context, namespace *Namespace) error {
+	var existingNamespace *Namespace
+	var err error
+	if len(namespace.Name) > 0 {
+		if existingNamespace, err = s.storageRepo.GetByName(ctx, namespace.Name); err != nil {
+			return err
+		}
+	} else {
+		if existingNamespace, err = s.storageRepo.GetByID(ctx, namespace.ID); err != nil {
+			return err
+		}
+	}
+	if len(namespace.State) > 0 {
+		existingNamespace.State = namespace.State
```

Comment:
We should not allow changing state till automation is not available for migrating one state to another.

Reply:
Agree so I have not added `namespace` update call in `cli` but I feel that instead of leaving this behind in the API, we can still have it to make it complete. What do you think?